### DISCUSSION TITLE
[appDir] Custom Error Boundary

### DISCUSSION THREAD

**Discussion by quick007** [vercel/next.js#44631] [Ideas]
### Describe the feature you'd like to request

One feature that's been missing is custom error boundaries. Right now, whenever there's an error within an app, it defaults to the closest `error.tsx` file. This can be all fine and dandy if the page you're editing looks like this: 
![image](https://user-images.githubusercontent.com/69743171/209425421-dc0bd402-cfdd-4a6c-8151-87a5582c714c.png)
In my case, I have a data-fetching request on the homepage. If that blows up, rather than being able to handle that one part of the UI, it goes to my nearest error boundary, `error.tsx`. Now my entire homepage is full of an error screen. 

### Describe the solution you'd like

Here's how this could be implemented:

```html
<ErrorBoundary fallback={<Error />}>
  <Component />
</ErrorBoundary>
```
Look familiar? 
![image](https://user-images.githubusercontent.com/69743171/209425597-52dc8aec-1cc5-4ab2-9852-d88fdc461f8a.png)

This feature would make it much easier to create custom, better-tailored errors

### Describe alternatives you've considered

N/A

**Comment by SukkaW** [comment]
You are not limited by Next.js' `error.tsx`. You can always bring your own error boundary, wrap your component in it, and put them inside the page!

https://reactjs.org/docs/error-boundaries.html

You can also install [react-error-boundary](https://www.npmjs.com/package/react-error-boundary) if you don't like to write the class component (the library does that for you).

**Comment by quick007** [comment]
I saw those docs, but assumed they were outdated. Great to know they're not! It would still be great to have a first-party solution for non-class components, but I'll definitely use that package for now.

**Reply by balazsorban44** [reply]
Curious, what gave you the impression it was outdated? The fact that it's a class component?
